pkg/servicemanager/manager.go

```diff
@@ -94,12 +94,13 @@ func (m *ServiceManager) Run(ctx context.Context, ready chan<- struct{}, stopped
 	return ctx.Err()
 }
 
-func (m *ServiceManager) asyncRun(ctx context.Context, service Service) (<-chan struct{}, <-chan struct{}) {
+func (m *ServiceManager) asyncRun(ctx context.Context, cancel context.CancelFunc, service Service) (<-chan struct{}, <-chan struct{}) {
 	ready, stopped := make(chan struct{}), make(chan struct{})
 	go func() {
 		klog.Infof("Starting %s", service.Name())
 		if err := service.Run(ctx, ready, stopped); err != nil {
```

Comment:
If the `Run()` call panics (either due to a panic called by our code, or by vendored dependent service) the cancel() won't occur.  It may be cleaner to exec the cancel in a `defer func(){  if r := recover(); r != nil { cancel(); //log the r string } }()` in a parent caller.  This should allow us to report fatal errors as fatal, shut down the services gracefully, and hopefully curb some of those 10k line stack traces.

Reply:
I like that. @fzdarsky was commenting on something similar but simpler that other projects are doing.

Reply:
Done, and verified with testing:

``` service-manager: context canceled
I0128 14:43:51.473440  197278 manager.go:115] Starting foo
I0128 14:43:51.473470  197278 manager.go:115] Starting bar-panic
E0128 14:43:52.474336  197278 manager.go:103] bar-panic crashed in panic: I'm in panic, stopping MicroShift
I0128 14:43:52.474368  197278 manager.go:120] foo completed ```